src/main/java/rs/hybridit/WebSecurityConfig.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import rs.hybridit.auth.JwtAuthenticationRequest;
+import rs.hybridit.model.Role;
+import rs.hybridit.model.User;
+import rs.hybridit.model.UserTokenState;
+import rs.hybridit.security.TokenHelper;
+import rs.hybridit.auth.TokenAuthenticationFilter;
+import rs.hybridit.service.UserService;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@AllArgsConstructor
+@Slf4j
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+	private TokenHelper tokenHelper;
+	private PasswordEncoder passwordEncoder;
+	private UserService userService;
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+
+		http
+			// communication between client and server is stateless
+			.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
+			.authorizeRequests()
+			.antMatchers("/auth/login").permitAll()
+			// every request needs to be authorized
+			.anyRequest().authenticated().and()
+			// add filter before every request
+			.addFilterBefore(new TokenAuthenticationFilter(tokenHelper, userService),
+				BasicAuthenticationFilter.class);
+		http.csrf().disable();
+
+	}
+
+	public void configure(WebSecurity web) throws Exception {
+		// Token Filter will ignore these paths
+		web.ignoring().antMatchers(HttpMethod.POST, "/auth/login", "/h2/**");
+		web.ignoring().antMatchers(HttpMethod.GET, "/", "/login", "/h2/**", "/webjars/**", "/*.html", "/favicon.ico",
+			"/**/*.html", "/**/*.css", "/**/*.js");
+	}
+
+
+	public UserTokenState login(JwtAuthenticationRequest authenticationRequest) throws Exception {
+		final Authentication authentication;
+		try {
+			authentication = authenticationManagerBean().authenticate(new UsernamePasswordAuthenticationToken(
+				authenticationRequest.getUsername(), authenticationRequest.getPassword()));
+		} catch (BadCredentialsException e) {
+			return null;
+		}
+		User user = (User) authentication.getPrincipal();
+		SecurityContextHolder.getContext().setAuthentication(authentication);
+		String jwt = tokenHelper.generateToken(user.getUsername());
+		int expiresIn = tokenHelper.getExpiredIn();
+		Role role = null;
+		if (user.getAuthority().getRole().equals(Role.ADMIN)) {
+			role = Role.ADMIN;
```

Comment:
You should use Role.valueOf(String) to compare string role name and enumeration

Reply:
But getRole() returns enumeration.